doc/sphinx/src/mesh/mesh.rst

```diff
@@ -50,3 +50,64 @@ member.
   time-integration advance. The default behavior calls to each package's
   (StateDesrcriptor's) ``PreStepDiagnostics`` method which, in turn,
   delegates to a ``std::function`` member that defaults to a no-op.
+
+Multi-grid Grids Stored in ``Mesh``
+-----------------------------------
+
+If the parameter ``parthenon/mesh/multigrid`` is set to ``true``, the ``Mesh``
+constructor and AMR routines populate both 
+``std::vector<LogicalLocMap_t> Mesh::gmg_grid_locs`` and 
+``std::vector<BlockList_t> gmg_block_lists``, where each entry into the vectors 
+describes one level of the of the geometric multi-grid (GMG) mesh. For refined 
+meshes, each GMG level only includes blocks that are at a given logical level 
+(starting from the finest logical level on the grid and including both internal 
+and leaf nodes in the refinement tree) as well as leaf blocks on the next coarser 
+level that are neighbors to finer blocks, which implies that below the root grid 
+level the blocks may not cover the entire mesh. For levels above the root grid, 
+blocks may change shape so that they only cover the domain of the root grid. Note 
+that leaf blocks may be contained in multiple blocklists, and the lists all point
+to the same block (not a separate copy).
+
+*GMG Implementation Note:*
+The reason for including two levels in the GMG block lists is for dealing with 
+accurately setting the boundary values of the fine blocks. Convergence can be poor 
+or non-exististent if the fine-coarse boundaries of a given level are not 
+self-consistently updated (since the boundary prolongation from the coarse grid to 
+the fine grid also depends on interior values of the fine grid that are being updated 
+by a smoothing operation). This means that each smoothing step, boundary communication 
+must occur between all blocks corresponding to all internal and leaf nodes at a given 
+level in the tree and with all leaf nodes at the next coarser level which abut blocks 
+at the current level. Therefore, the GMG block lists contain blocks at two levels, but 
+smoothing operations etc. should usually only occur on the subset of those blocks that 
+are at the fine level.
+
+To work with these GMG levels, ``MeshData`` objects containing these blocks can 
+be recovered from a ``Mesh`` pointer using 
+
+.. code:: c++
+  auto &md = pmesh->gmg_mesh_data[level].GetOrAdd(level, "base", i);
+
+This ``MeshData`` will include blocks at the current level and possibly some 
+blocks at the next coarser level. Often, one will only want to operate on blocks
+on the finer level (the coarser blocks are required mainly for boundary 
+communication). To make packs containing only a subset of blocks from a 
+GMG ``MeshData`` pointer ``md``, one would use 
+
+.. code:: c++
+  int nblocks = md->NumBlocks();
+  std::vector<bool> include_block(nblocks, true);
+  for (int b = 0; b < nblocks; ++b)
+    include_block[b] =
+        (md->grid.logical_level == md->GetBlockData(b)->GetBlockPointer()->loc.level());
```

Comment:
After working with this for a while, I am more convinced that getting only the blocks selected in the code snippet above for gmg mesh data should be the default behavior of the packs (since you almost always only want the finer blocks in a two-level composite gmg grid).

Reply:
Resolving this and putting off to another PR, see to-dos in #950.